include/dsn/c/api_layer1.h

```diff
@@ -541,26 +542,25 @@ typedef enum dsn_msg_parameter_type_t
     MSG_PARAM_NONE = 0,           ///< nothing  
 } dsn_msg_parameter_type_t;
 
-enum dsn_msg_serialize_format
+typedef enum dsn_msg_serialize_format
 {
-    DSF_THRIFT_BINARY,
-    DSF_THRIFT_COMPACT,
-    DSF_THRIFT_JSON,
-    DSF_PROTOC_BINARY,
-    DSF_PROTOC_JSON,
-
-    DSF_COUNT,
-    DSF_INVALID
-};
+    DSF_INVALID = 0,
+    DSF_THRIFT_BINARY = 1,
+    DSF_THRIFT_COMPACT = 2,
+    DSF_THRIFT_JSON = 3,
+    DSF_PROTOC_BINARY = 4,
+    DSF_PROTOC_JSON = 5,
+} dsn_msg_serialize_format;
 
 /*! RPC message context */
 typedef union dsn_msg_context_t
 {
     struct {
         uint64_t is_request : 1;           ///< whether the RPC message is a request or response
         uint64_t is_forwarded : 1;         ///< whether the msg is forwarded or not
-        uint64_t unused : 5;               ///< not used yet
+        uint64_t unused : 4;               ///< not used yet
         uint64_t serialize_format : 4;     ///< dsn_msg_serialize_format
+        uint64_t is_forward_disabled: 1;   ///< whether forward a message to real leader
```

Comment:
I mean that "is_forward_disabled"  does not really have the same scope compared to "is_forward_not_supported". Using the latter prevents questions as the above I asked, because the code is more self-explained in that case.